src/corehost/cli/apphost/bundle/bundle_runner.cpp

```diff
@@ -312,21 +312,41 @@ StatusCode bundle_runner_t::extract()
         }
 
         // Commit files to the final extraction directory
-        if (pal::rename(m_working_extraction_dir.c_str(), m_extraction_dir.c_str()) != 0)
+        // Retry the move operation with some wait in between the attempts. This is to workaround possible file locking
+        // caused by AV software. Basically the extraction process above just write a bunch of executable files to disk
+        // and some AV software may decide to scan it on write. If this happens the files will be locked and we won't
+        // be allowed to move them around.
+        int retry_count = 500;
+        while (true)
         {
-            if (can_reuse_extraction())
+            if (pal::rename(m_working_extraction_dir.c_str(), m_extraction_dir.c_str()) != 0)
             {
-                // Another process successfully extracted the dependencies
-
-                trace::info(_X("Extraction completed by another process, aborting current extracion."));
-
-                remove_directory_tree(m_working_extraction_dir);
-                return StatusCode::Success;
+                bool should_retry = errno == EACCES;
+                if (can_reuse_extraction())
+                {
+                    // Another process successfully extracted the dependencies
+
+                    trace::info(_X("Extraction completed by another process, aborting current extracion."));
+
+                    remove_directory_tree(m_working_extraction_dir);
+                    return StatusCode::Success;
```

Comment:
Should we be calling `fclose(m_bundle_stream)` here? All other cases cause that to happen.

Reply:
Good catch. Yes, seems like it, even though the original code didn't do it in this path.